doc/getting_started/toolchain_3rd_party_x_compilers.rst

```diff
@@ -0,0 +1,128 @@
+.. _third_party_x_compilers:
+
+3rd Party Toolchains
+####################
+
+A "3rd party toolchain" is an officially supported toolchain provided by an
+external organization. Several of these are available.
+
+GNU ARM Embedded
+****************
+
+#. Download and install a `GNU ARM Embedded`_ build for your operating system
+   and extract it on your file system.
+
+   .. warning::
+
+      Like the Zephyr repository, do not install the toolchain in a directory
+      with spaces anywhere in the path.
+
+#. Configure the environment variables needed to inform the Zephyr build system
+   to use this toolchain:
+
+   - Set :envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to ``gnuarmemb``.
+   - Set :envvar:`GNUARMEMB_TOOLCHAIN_PATH` to the toolchain installation
+     directory.
+
+   For example, if you installed to :file:`<user folder>/gnu_arm_embedded` (or
+   ``/c/gnuarmemb`` in the MSYS2 environment on Windows):
+
+   .. code-block:: console
+
+      # Linux or macOS
+      export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
+      export GNUARMEMB_TOOLCHAIN_PATH="~/gnu_arm_embedded"
+
+      # Windows in cmd.exe
+      set ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
+      set GNUARMEMB_TOOLCHAIN_PATH="%userprofile%\gnu_arm_embedded"
+
+      # Windows in an MSYS2 shell
+      export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
+      export GNUARMEMB_TOOLCHAIN_PATH=/c/gnuarmemb
+
+Intel ISSM
+**********
+
+.. note::
+
+   The ISSM toolset only supports development for Intel |reg| Quark |trade|
+   Microcontrollers, for example, the Arduino 101 board.  (Check out the
+   "Zephyr Development Environment
+   Setup" in this `Getting Started on Arduino 101 with ISSM`_ document.)
+   Additional setup is required to use the ISSM GUI for development.
+
+#. Install the ISSM toolchain by downloading it from the Intel Developer Zone's
+   `ISSM Toolchain`_ downloads page, then extracting the archive somewhere on
+   your system.
+
+   .. warning::
+
+      Like the Zephyr repository, do not install the toolchain in a directory
+      with spaces anywhere in the path.
+
+#. Configure the environment variables needed to inform the Zephyr build system
+   to use this toolchain:
+
+   - Set :envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to ``issm``.
+   - Set :envvar:`ISSM_INSTALLATION_PATH` to the directory containing the
+     extracted files.
+
+   For example:
+
+   .. code-block:: console
+
+      # Linux
+      export ZEPHYR_TOOLCHAIN_VARIANT=issm
+      export ISSM_INSTALLATION_PATH=/home/you/Downloads/issm0-toolchain-windows-2017-02-07
+
+      # Windows in cmd.exe
+      set ZEPHYR_TOOLCHAIN_VARIANT=issm
+      set ISSM_INSTALLATION_PATH=c:\issm0-toolchain-windows-2017-01-25
+
+      # Windows in an MSYS2 shell
```

Comment:
I would remove this section. We should really deprecate MSYS2, so including additional references to it should be discouraged.